### PR TITLE
Aj/static call VAN-170

### PR DESCRIPTION
**Pull request by veriAjinkya**
Adds translation of static call. 
Adds testcase from bug report

### CODE REVIEW

File: lib/libyul2llvm/YulASTVisitor/IntrinsicRewriter.cpp [hunk 1]
```diff
@@ -172,6 +172,66 @@ void YulIntrinsicHelper::rewriteCallIntrinsic(llvm::CallInst *callInst) {
...
+    llvm::CallInst *newCall =
+        llvm::CallInst::Create(callF, args, "ret_struct", callInst);
+    removeOldCallArgs(callInst);
+    adjustCallReturns(callInst, newCall, retStructType, visitor);
+  } else {
+    llvm::CallInst *newCall = llvm::CallInst::Create(callF, args, "call_rv");
+    removeOldCallArgs(callInst);
+    llvm::ReplaceInstWithInst(callInst, newCall);
+  }
+}
```
Comment (Technius) [inline lib/libyul2llvm/YulASTVisitor/IntrinsicRewriter.cpp:233]: This largely looks similar to `rewriteCallIntrinsic`. Is it possible to refactor this into a helper function and reuse the code?

File: tests/e2e/FlashLoan-runner.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+CHECK: store i256 %"convert_t_contract$_IERC20_$100_to_t_address6", i256* %addrPtr, align 4
+CHECK: %valuePtr = getelementptr %ExtCallContextType, %ExtCallContextType* %_0x70a08231ctx, i32 0, i32 2
+CHECK: store i256 0, i256* %valuePtr, align 4
+CHECK: %bufferPtr = getelementptr %ExtCallContextType, %ExtCallContextType* %_0x70a08231ctx, i32 0, i32 3
+CHECK: store i256* %ret_buffer, i256** %bufferPtr, align 8
+CHECK: %retLenPtr = getelementptr %ExtCallContextType, %ExtCallContextType* %_0x70a08231ctx, i32 0, i32 4
+CHECK: store i256 32, i256* %retLenPtr, align 4
+CHECK: %ret_struct = call %_0x70a08231_statusRetType* @pyul_staticcall_0x70a08231(i256 addrspace(1)* %__self, %ExtCallContextType* %_0x70a08231ctx, i256 %caller7)
+CHECK: %ptr_status = getelementptr %_0x70a08231_statusRetType, %_0x70a08231_statusRetType* %ret_struct, i32 0, i32 0
+CHECK: %status = load i256, i256* %ptr_status, align 4
```
Comment (Technius) [inline tests/e2e/FlashLoan-runner.txt:16]: This test should also check that the `pyul_call` call for `token.transfer()` is emitted.

File: lib/libyul2llvm/YulASTVisitor/IntrinsicRewriter.cpp (hunk 1 above)
Comment (veriAjinkya) [reply lib/libyul2llvm/YulASTVisitor/IntrinsicRewriter.cpp:233]: This is done

File: tests/e2e/FlashLoan-runner.txt (hunk 2 above)
Comment (veriAjinkya) [reply tests/e2e/FlashLoan-runner.txt:16]: done

**Review by Technius — APPROVED** [review]
LGTM. I have a few more refactoring ideas to improve robustness, but we can discuss those offline so we don't block this PR.

**Pull request merged**